### PR TITLE
Minio integration with Object store

### PR DESCRIPTION
**Pull request by aligeti**
This change is [Reviewable](https://reviewable.io/reviews/storj/storj/156)

### CODE REVIEW

**Comment by kaloyan-raev** [comment]
@aligeti How about adding tests with mocked `objects.Store` interface?

File: pkg/miniogw/gateway-storj.go [hunk 1]
```diff
@@ -95,12 +95,12 @@ func (s *storjObjects) GetObject(ctx context.Context, bucket, object string,
...
 	defer mon.Task()(&ctx)(&err)
 	objpath := paths.New(bucket, object)
-	rr, _, err := s.storj.os.GetObject(ctx, objpath)
+	rr, _, err := s.storj.os.Get(ctx, objpath)
 	if err != nil {
 		return err
 	}
 	defer func() {
-		if err := rr.Close(); err != nil {
+		if err = rr.Close(); err != nil {
```
Comment (jtolds) [inline pkg/miniogw/gateway-storj.go]: oh man this is a huge bug. i almost want to figure out how to get rid of errcheck for these defer issues.

it turns out that the colon you removed was load bearing. if you have the colon, inside the defer you're defining a new err variable and checking it. without the colon, you're now overwriting the return value, which is named err!!

summary: with a colon, this is fine. without a colon, this is a huge bug that's going to cause very hard to debug silent error swallowing.

@coyle this is a huge footgun. what do you think? should we try and get errcheck to ignore deferred calls?

Comment (kaloyan-raev) [reply pkg/miniogw/gateway-storj.go]: @aleitner defined `utils.Close()` helper in PR #134: https://github.com/storj/storj/pull/134/files#diff-a0fc870d3e98cb9e7ffa9fcb9fb1d041R39

Once it's merge we can just call `defer utils.Close(rr)` here.

**Comment by kaloyan-raev** [comment]
I think it's only the tests to be finished. And the conflict to be resolved.

**Pull request merged**